tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/WhatisResponse.java

```diff
@@ -0,0 +1,65 @@
+package org.unicode.cldr.web.api;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.unicode.cldr.util.CLDRLocale;
+
+public final class WhatisResponse {
+    @Schema(description = "query string")
+    public String q;
+    public String q2 = null;
+    public String err = null;
+    public List<XpathResult> xpath = null;
+    public List<StandardCodeResult> exact = null;
+    public List<StandardCodeResult> full = null;
+    public int pathsSearched = 0;
+    public int localesSearched = 0;
+
+    public WhatisResponse(String queryString) {
+        q = queryString;
+    }
+
+    class XpathResult {
+        public String loc;
+        public String name;
+        public List<PathAndValue> matches = new ArrayList<>();
+
+        public XpathResult(CLDRLocale locale) {
+            loc = locale.toString();
+            name = locale.getDisplayName();
+        }
+    }
+
+    class PathAndValue {
+        public String path;
+        public String value;
+
+        public PathAndValue(String p, String v) {
+            path = p;
+            value = v;
+        }
+    }
+
+    class StandardCodeResult {
+        public String type;
+        public String code;
+        public List<StandardCodeResultDataItem> data = new ArrayList<>();
```

Comment:
@btangmu this also needs to be an object array

Reply:
> this also needs to be an object array

Yes, I made all the List members private with getters; still the same error. Here's the latest:
```
package org.unicode.cldr.web.api;

import java.util.ArrayList;
import java.util.List;

import org.eclipse.microprofile.openapi.annotations.media.Schema;
import org.unicode.cldr.util.CLDRLocale;

public final class WhatisResponse {
    @Schema(description = "query string")
    public String q;
    public String q2 = null;
    public String err = null;
    public int pathsSearched = 0;
    public int localesSearched = 0;

    private List<XpathResult> xpath = null;
    private List<StandardCodeResult> exact = null;
    private List<StandardCodeResult> full = null;

    public WhatisResponse(String queryString) {
        q = queryString;
    }

    public XpathResult[] getXpath() {
        System.out.println("Hello my name is getXpath");
        return xpath.toArray(new XpathResult[xpath.size()]);
    }

    public StandardCodeResult[] getExact() {
        System.out.println("Hello my name is getExact");
        return exact.toArray(new StandardCodeResult[exact.size()]);
    }

    public StandardCodeResult[] getFull() {
        System.out.println("Hello my name is getFull");
        return full.toArray(new StandardCodeResult[full.size()]);
    }

    class XpathResult {
        public String loc;
        public String name;
        private List<PathAndValue> matches = new ArrayList<>();

        public XpathResult(CLDRLocale locale) {
            loc = locale.toString();
            name = locale.getDisplayName();
        }

        public PathAndValue[] getMatches() {
            System.out.println("Hello my name is getMatches");
            return matches.toArray(new PathAndValue[matches.size()]);
        }

        public void addMatch(PathAndValue pathAndValue) {
            matches.add(pathAndValue);
        }
    }

    class PathAndValue {
        public String path;
        public String value;

        public PathAndValue(String p, String v) {
            path = p;
            value = v;
        }
    }

    class StandardCodeResult {
        public String type;
        public String code;
        private List<StandardCodeResultDataItem> data = new ArrayList<>();

        public StandardCodeResult(String t, String c) {
            type = t;
            code = c;
        }

        public StandardCodeResultDataItem[] getData() {
            System.out.println("Hello my name is getData");
            return data.toArray(new StandardCodeResultDataItem[data.size()]);
        }

        public void addData(StandardCodeResultDataItem item) {
            data.add(item);
        }
    }

    class StandardCodeResultDataItem {
        public String s;
        public boolean isWinner;

        public StandardCodeResultDataItem(String str, boolean win) {
            s = str;
            isWinner = win;
        }
    }

    public void addExact(StandardCodeResult r) {
        if (exact == null) {
            exact = new ArrayList<>();
        }
        exact.add(r);
    }

    public void addFull(StandardCodeResult r) {
        if (full == null) {
            full = new ArrayList<>();
        }
        full.add(r);
    }

    public void addXpath(XpathResult r) {
        if (xpath == null) {
            xpath = new ArrayList<>();
        }
        xpath.add(r);
    }
}
```

I wonder if there's a debug mode for the serialization library, or a way to step through its code, to get more clues about where things go wrong. I'd also love to find some documentation clarifying the requirements for serialization, such as when does public/private actually matter, when are `@Schema` or other annotations or metadata required.

This article might have useful ideas: https://www.baeldung.com/jackson-custom-serialization

It's not even clear to me what the relevant technology is here: Jackson? Swagger?

Thanks!